service/history/execution/history_builder.go

```diff
@@ -576,6 +576,7 @@ func (b *HistoryBuilder) newWorkflowExecutionStartedEvent(
 
 	parentInfo := startRequest.ParentExecutionInfo
 	if parentInfo != nil {
+		attributes.ParentWorkflowDomainID = parentInfo.DomainUUID
```

Comment:
Not related to this PR except by proximity, but:
This kind of code:
```go
	attributes := &types.WorkflowExecutionStartedEventAttributes{}
	attributes.WorkflowType = request.WorkflowType
	attributes.TaskList = request.TaskList
	...
```
Is rather hard to lint, e.g. that fields aren't forgotten.  Or to be confident of it while coding.
It currently looks complete, but may be worth rewriting to a:
```go
	attributes := &types.WorkflowExecutionStartedEventAttributes{
		WorkflowType:                        request.WorkflowType,
		TaskList:                            request.TaskList,
		Header:                              request.Header,
		Input:                               request.Input,
		ExecutionStartToCloseTimeoutSeconds: common.Int32Ptr(*request.ExecutionStartToCloseTimeoutSeconds),
```
Initializer (with an if for parentinfo stuff, since there's not much choice there).

As a concrete example, I can ctrl-space in that initializer and immediately see that 3 fields are missing - the parentinfo ones, handled below.  I have yet to see tooling that helps even _that_ much with the field-assignment pattern.

---

I've got a 90%-complete linter that checks for "complete" struct initializations, and it wouldn't be too hard to include "this func contains an assignment for every field" info in that check too.  Maybe it'd be worth finishing and bringing in?  Incomplete-initialization has been a fairly large source of sometimes-minor, often-long-lasting data losses in my experience.

Reply:
Yes, I do prefer direct initialization as well. Updated it.

If we could lint on that, it would be amazing. Just wondering how it is handled when some fields are intentionally left out. Do we add annotation in the code?